imports/api/employee.js

```diff
@@ -0,0 +1,234 @@
+import { Meteor } from "meteor/meteor";
+import { Employees, Tasks } from "./db";
+import EMPLOYEESAPI from "../constant/methods/employeesAPI";
+import { isAuthenticated } from "../util/authUtil";
+import { AuthError, EmployeeError } from "../constant/error";
+import { EmployeeMessage } from "../constant/message";
+import { removeElement, addToList } from "../util/arrayUtil";
+import { isAdmin, removeRelatedTasks } from "../util/employeeUtil";
+
+if (Meteor.isServer) {
+	Meteor.publish("employees", () => Employees.find());
+}
+
+export const insertEmployee = (
+	db,
+	_id,
+	employeeId,
+	firstName,
+	lastName,
+	role,
+) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+
+	return db.insert(
+		{
+			employeeId,
+			firstName,
+			lastName,
+			onBoard: new Date(),
+			// keep it null for now
+			managerId: null,
+			teamId: null,
+			tasksAssienedId: [],
+			tasksCreatedId: [],
+			tasksWatchingId: [],
+			requestsSentId: [],
+			requestReceivedId: [],
+			commentsId: [],
+			filesId: [],
+			employeesId: [],
+			individualsId: [],
+			teamsId: [],
+			// role should be one of"employer", "employee", "admin"?
+			role,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_INSERT_FAIL);
+			} else {
+				console.log(EmployeeMessage.EMPLOYEE_CREATED);
+			}
+		},
+	);
+};
+
+export const removeEmployee = (db, _id, employeeId) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+	const employee = Employees.findOne({ employeeId });
+	const removedId = employee.employeeId;
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	if (removedId !== this.employeeId) {
```

Comment:
For checking admin removing self, just do (removedId !== _id). "this" has no context here.

Reply:
i think neither `removedId !== _id` nor `removedId !== this.employeeId` is good in this case. since either `_id` and `employeeid` means the id of admin, which are supposed to not equal to the one that need to be removed. I think this check does not make sense for privilege error checking.

Reply:
maybe we should check `removedId === this.employeeId` instead?